udata/frontend/markdown.py

```diff
@@ -101,7 +101,7 @@ def mdstrip(value, length=None, end='…'):
     rendered = md(value)
     text = do_striptags(rendered)
     if length > 0:
-        text = do_truncate(text, length, end=end)
+        text = do_truncate(text, length, end=end, leeway=2)
```

Comment:
I think the filter signature need to be updated to reflect jinja 2.9 changes (and make tests pass):
- decorate the method with `@environmentfilter` and add `env` as first parameter (see https://github.com/pallets/jinja/blob/master/jinja2/filters.py#L466-L467)
- pass `env` as first parameter to `do_truncate`